opentracing/scope.py

```diff
@@ -78,5 +78,7 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         and added as a tag to the :class:`Span`.
         :attr:`~operation.ext.tags.ERROR` will also be set to `True`.
         """
-        self.span._on_error(exc_type, exc_val, exc_tb)
+        span = self.span
+        if span:
+            span._on_error(exc_type, exc_val, exc_tb)
```

Comment:
hm, this looks like a bad pattern altogether - the API does not require the span to have `_on_error` method, it should really be a static method on the Span class that takes span as argument, that way it is safe to be reused even with Noop span.

Reply:
I just made corresponding changes.
Please re-review.
Thanks!